### PR TITLE
Harden fdns.service

### PR DESCRIPTION
**Pull request by rusty-snake**
Possible improvements:
 - extend the `SystemCallFilter=~` filter
 - minimize "`private-etc`"

Tested distros:
 - [x] Fedora
 - [ ] Debian
 - [ ] Arch

### CODE REVIEW

**Comment by netblue30** [comment]
all set, thanks!

File: etc/fdns.service [hunk 1]
```diff
@@ -21,5 +21,98 @@ Restart=on-failure
...
+PrivateTmp=true
+
+# FJ: --private-dev --no…
+PrivateDevices=true
+DevicePolicy=closed
+
+# FJ: --caps.keep=kill,net-bind,setgid,setuid,sys-admin,sys-chroot
+CapabilityBoundingSet=CAP_KILL CAP_NET_BIND_SERVICE CAP_SETGID CAP_SETUID CAP_SYS_ADMIN CAP_SYS_CHROOT
+
+# FJ: --memor-deny-write-execute --nonewprivs --protocol=unix,inet,inet6
```
Comment (glitsj16) [inline etc/fdns.service]: Small typo: --memor**y**-deny-write-execute

File: etc/fdns.service [hunk 2]
```diff
@@ -21,5 +21,98 @@ Restart=on-failure
...
+MemoryDenyWriteExecute=true
+NoNewPrivileges=true
+RestrictAddressFamilies=AF_UNIX AF_INET AF_INET6
```
Comment (glitsj16) [inline etc/fdns.service]: This breaks on Arch with systemd 244.1-1:
RestrictAddressFamilies=AF_UNIX AF_INET AF_INET6

File: etc/fdns.service [hunk 3]
```diff
@@ -21,5 +21,98 @@ Restart=on-failure
...
+
+# FJ: --seccomp.keep=@system-service,@mount,seccomp --seccomp.drop=@debug,@obsolete --seccomp.block-secondary
+SystemCallFilter=@system-service @mount seccomp
```
Comment (glitsj16) [inline etc/fdns.service]: SystemCallFilter is not supported on Ubuntu 16.04 LTS, which has systemd [229-4ubuntu21.23]

File: etc/fdns.service [hunk 4]
```diff
@@ -21,5 +21,98 @@ Restart=on-failure
 #    and change the service type from `forking' to `simple'.
 #StandardOutput=append:/tmp/fdns-log.txt
 
+AmbientCapabilities=
+LockPersonality=true
```
Comment (glitsj16) [inline etc/fdns.service]: The following options are unsupported on Ubuntu 16.04 LTS with systemd 229-4ubuntu21.23:

LockPersonality
RestrictNamespaces
RestrictRealtime
RestrictSUIDSGID
ProtectHostname
PrivateMounts

ProtectKernelTunables
ProtectKernelModules
ProtectControlGroups

File: etc/fdns.service (hunk 1 above)
Comment (rusty-snake) [reply etc/fdns.service]: Fixed 285de74c.

File: etc/fdns.service (hunk 2 above)
Comment (rusty-snake) [reply etc/fdns.service]: Fixed 285de74c.

File: etc/fdns.service (hunk 4 above)
Comment (rusty-snake) [reply etc/fdns.service]: Is anything broken or just not supported? On Fedora 30 is `ProtectHostname` also not supported, but it is just logged and ignored.

Comment (glitsj16) [reply etc/fdns.service]: These are indeed unsupported but not breaking anything. Just log pollution. Thanks for the work on this BTW!

**Pull request merged**